ampd/src/cosmos.rs

```diff
@@ -59,9 +61,14 @@ pub struct CosmosGrpcClient {
 }
 
 impl CosmosGrpcClient {
-    pub async fn new(url: &str) -> Result<Self> {
+    pub async fn new(url: &str, timeout: Duration) -> Result<Self> {
         let endpoint: tonic::transport::Endpoint = url.parse().map_err(ErrorExt::into_report)?;
-        let conn = endpoint.connect().await.map_err(ErrorExt::into_report)?;
+        let conn = endpoint
+            .timeout(timeout)
+            .connect_timeout(timeout)
```

Comment:
what's the difference, one is for establishing and one is for individual requests?

Reply:
Correct. I figure there's no reason to have both being configurable separately at the moment, but maybe in the future